packages/broker/test/unit/ConfigWizard.test.ts

```diff
@@ -1,178 +1,207 @@
-        expect(wallet.address.charAt(1)).toBe('x')
+        it ('invalid data: out-of-range number', () => {
+            const validate = portPrompt.validate!
+            const port = '10000000000'
+            expect(validate(port)).toBe(`Out of range port ${port} provided (valid range 1024-49151)`)
+        })
+
+        it ('invalid data: non-numeric', () => {
+            const validate = portPrompt.validate!
+            const port = 'Not A Number!'
+            expect(validate(port)).toBe(`Non-numeric value provided`)
+        })
     })
 
-    it ('should import a valid privateKey', async() => {
-        const privateKey = '0x8223c67c3c430b75931a4a1620434208749120f7d37204e93bf877e5014499ed'
-        mockPromptMethod(wizard, {
-            generateOrImportEthereumPrivateKey: 'import',
-            privateKey
+    describe('storagePrompt validation', () => {
+        let tmpDataDir: string
+
+        beforeAll(() => {
+            tmpDataDir = mkdtempSync(path.join(os.tmpdir(), 'broker-test-config-wizard'))
+        })
+
+        it ('happy path', () => {
+            const validate = selectDestinationPathPrompt.validate!
+            const validPath = tmpDataDir + '/test-config.json'
+            expect(validate(validPath)).toBe(true)
+        })
+
+        it ('happy path with overwrite destination', () => {
+            const validate = selectDestinationPathPrompt.validate!
+            const validPath = tmpDataDir + '/test-config.json'
+            writeFileSync(validPath, JSON.stringify({}))
+            const answers: any = {}
+            const isValid = validate(validPath, answers)
+            expect(isValid).toBe(true)
+            expect(answers.parentDirExists).toBe(true)
+            expect(answers.fileExists).toBe(true)
         })
 
-        const importedWallet: Wallet = await wizard.generateOrImportPrivateKey()
-        expect(importedWallet.privateKey).toEqual(privateKey)
+        it ('invalid path provided', () => {
+            const validate = selectDestinationPathPrompt.validate!
+            const invalidPath = `/invalid-path/${Date.now()}`
+            const answers: any = {}
+            const isValid = validate(invalidPath, answers)
+            expect(isValid).toBe(false)
+            expect(answers.parentDirExists).toBe(false)
+            expect(answers.fileExists).toBe(false)
+
+        })
     })
 
-    it ('should throw when importing an invalid privateKey', async() => {
-        const privateKey = '0xNotAValidPrivateKey'
-        try {
-            mockPromptMethod(wizard, {
-                generateOrImportEthereumPrivateKey: 'import',
-                privateKey
+    describe('createStorageFile', () => {
+        let tmpDataDir: string
+        let config: any
+
+        beforeAll(() => {
+            tmpDataDir = mkdtempSync(path.join(os.tmpdir(), 'broker-test-config-wizard'))
+        })
+
+        beforeEach(() => {
+            config = { ... DEFAULT_CONFIG, plugins: { ... DEFAULT_CONFIG.plugins } }
```

Comment:
Why is this needed?

Reply:
Assumed that the simple assignation was not alright since it had to be changed as well in code.

Reply:
But the value is assigned only in beforeEach, so it is a constant (and if you don't change the contents in any way, why to make a copy)

Reply:
So can we revert to `config = DEFAULT_CONFIG`?

Reply:
`createStorageFile` is the method which we are testing here. It accepts config: any. So an empty JSON is a valid input (if no need to be more complicated, the simplest input is the best)

Reply:
Terrific, implemented on the next commit!